CHTCollectionViewWaterfallLayout.m

```diff
@@ -423,7 +423,9 @@ - (UICollectionViewLayoutAttributes *)layoutAttributesForSupplementaryViewOfKind
 - (NSArray *)layoutAttributesForElementsInRect:(CGRect)rect {
   NSInteger i;
   NSInteger begin = 0, end = self.unionRects.count;
-  NSMutableArray *attrs = [NSMutableArray array];
+  NSMutableDictionary *cellAttrDict = [NSMutableDictionary dictionary];
+  NSMutableDictionary *supplAttrDict = [NSMutableDictionary dictionary];
+  NSMutableDictionary *decorAttrDict = [NSMutableDictionary dictionary];
```

Comment:
The problem is `indexPath`s can be the same and collide between each of those three object categories. I actually tried that approach on the first commit but it didn't work because of that. Items were overwritten by decorators or supplemental views and viceversa.